gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb

```diff
@@ -31,10 +31,8 @@ class MultipartFileUploader
       )
 
       # @option options [Client] :client
-      # @option options [Integer] :thread_count (THREAD_COUNT)
       def initialize(options = {})
         @client = options[:client] || Client.new
-        @thread_count = options[:thread_count] || THREAD_COUNT
```

Comment:
I think we still want to set `@thread_count` here - otherwise this could be a breaking change.  However - I think we can skip setting a default ie, just `@thread_count = options[:thread_count]`.

Then in the `upload_in_threads` method we can do something like:
```ruby
      def upload_in_threads(pending, completed, options)
        # code skipped for brevity here
     
        (@thread_count || options[:thread_count] || THREAD_COUNT).times do
          thread = Thread.new do
```

Reply:
Edit: Sorry, for the initial suggestion - I do agree with @mullermp's suggestion.  We can keep the defaulting of `@thread_count` in the initializer and use an `options.fetch(:thread_count, @thread_count)` in the `upload_in_threads` method.  I agree that preserves the normal order of precedence better.

Reply:
👍 Addressed in my latest commit @mullermp @alextwoods.

For consistency, I elected to implement the same variable precedence in `MultipartStreamUploader` as well.